bin/cheminfo-publish.js

```diff
@@ -23,245 +22,272 @@ const util = require('../src/util');
-    }
-
-    // Execute the tests
-    console.log('Running the tests');
-    yield execNpm('run test');
-
-    // Bump version
-    console.log('Bumping version');
-    const newVersion = bumpVersion[bump];
-    const newVersionString = `"version": "${newVersion}"`;
-    const versionReg = /"version": "[^"]+"/;
-
-    let packData = yield fs.readFile(packageJSONPath, 'utf8');
-    packData = packData.replace(versionReg, newVersionString);
-    yield fs.writeFile(packageJSONPath, packData);
-
-    if (hasPackageLock) {
-        let packLockData = yield fs.readFile(packageLockPath, 'utf8');
-        packLockData = packLockData.replace(versionReg, newVersionString);
-        yield fs.writeFile(packageLockPath, packLockData);
-    }
-
-    // Add/update changelog
-    yield updateHistory();
-
-    // Commit the update and tag it
-    yield child_process.exec('git add package.json History.md' + (hasPackageLock ? ' package-lock.json' : ''));
-    yield child_process.exec(`git commit -m ${newVersion}`);
-    yield child_process.exec(`git tag -a v${newVersion} -m v${newVersion}`);
-
-    // Publish package
-    console.log('Publishing package');
-    var publishOutput;
-    try {
-        publishOutput = yield execNpm('publish');
-    } catch (e) {
-        errorLog('npm publish failed, rolling back commits and tags');
-        yield child_process.exec(`git tag -d v${newVersion}`);
-        yield child_process.exec('git reset --hard HEAD~1');
-        return;
-    } finally {
-        log(publishOutput);
-    }
-
-    // Add to organization
-    var packages;
+    const confirm = (yield inquirer.prompt({
+      type: 'confirm',
+      name: 'c',
+      message: 'Are you sure',
+      default: false
+    })).c;
+    if (!confirm) return;
+  }
+
+  // Execute the tests
+  console.log('Running the tests');
+  yield execNpmStdout('run', 'test');
+
+  // Bump version
+  console.log('Bumping version');
+  const newVersion = bumpVersion[bump];
+  const newVersionString = `"version": "${newVersion}"`;
+  const versionReg = /"version": "[^"]+"/;
+
+  let packData = yield fs.readFile(packageJSONPath, 'utf8');
+  packData = packData.replace(versionReg, newVersionString);
+  yield fs.writeFile(packageJSONPath, packData);
+
+  if (hasPackageLock) {
+    let packLockData = yield fs.readFile(packageLockPath, 'utf8');
+    packLockData = packLockData.replace(versionReg, newVersionString);
+    yield fs.writeFile(packageLockPath, packLockData);
+  }
+
+  // Add/update changelog
+  yield updateHistory();
+
+  // Commit the update and tag it
+  const filesToAdd = ['package.json', 'History.md'];
+  if (hasPackageLock) {
+    filesToAdd.push('package-lock.json');
+  }
+  yield execa('git', ['add', ...filesToAdd]);
+  yield execa('git', ['commit', '-m', newVersion]);
+  yield execa('git', ['tag', '-a', `v${newVersion}`, '-m', `v${newVersion}`]);
+
+  // Publish package
+  console.log('Publishing package');
+  var publishOutput;
+  try {
+    throw new Error('stop here');
```

Comment:
wut?

Reply:
oups, I wanted to test it without publishing